File: Instructor Java/Week02/week 2/src/main/java/case01/HelloWorld.java
package case01;

import static java.text.MessageFormat.*;
import static java.util.Objects.isNull;
import static java.lang.System.out;

/**
 * @author: Waleed A.K.
 *
 */
public class HelloWorld {

  private String name;

  public void printHelloWorld() {
    String display = "Hello World --\tprintHelloWorld";
    out.println(display);
  }

  public void printHelloWorld2() {
    var display = "Hello World --\tprintHelloWorld2";
    out.println(display);
  }

  public void printHelloWithInput(String input) {
    var display = format("Hello {0} --\tprintHelloWithInput", input);
    out.println(display);
    var display2 = String.format("Hello %s --\tprintHelloWithInput", input);
    out.println(display2);
  }

  public void printHelloHandleField() {
    var input = name;

    if (isNull(input)) {
      input = "N/A";
    }
    var display = format("Hello {0} --\tprintHelloWithInput", input);
    out.println(display);
  }

  public void name(String name) { // set
    this.name = name;
  }

  public String getName() { // get
    return this.name;
  }
}

File: Instructor Python/Week03 Python/src/case01/HelloWorld.py
class HelloWorld():
    def __init__(self):
        self.__name: str = None

    def printHelloWorld(self):
        print('hello world\t--printHelloWorld')

    def printHelloWithInput(self, input):
        print(f'Hello {input}\t--printHelloWithInput')

    def printHelloHandleField(self):
        input = self.__name
        if input is None:
            input = "N/A"
        print(f"Hello {input}\t--printHelloHandleField")

    def __setName(self, value):
        self.__name = value

    def __getName(self):
        return self.__name

    name: str = property(__getName, __setName)


def Add(a, b):
    return a+b
